avatar/cases/le_host_test.py

```diff
@@ -139,6 +146,86 @@ def test_scan(
             scan.cancel()
             advertise.cancel()
 
+    @avatar.parameterized(
+        *itertools.product(
+            # The advertisement cannot be both connectable and scannable.
+            ('connectable', 'non_connectable', 'non_connectable_scannable'),
+            ('directed', 'undirected'),
+            # Bumble does not send multiple HCI commands, so it must also fit in
+            # 1 HCI command (max length 251 minus overhead).
+            (0, 150),
+            (PRIMARY_1M, PRIMARY_CODED),
+        ),
+    )  # type: ignore[misc]
+    def test_extended_scan(
+        self,
+        connectable_scannable: Union[
+            Literal['connectable'], Literal['non_connectable'], Literal['non_connectable_scannable']
+        ],
+        directed: Union[Literal['directed'], Literal['undirected']],
+        data_len: int,
+        primary_phy: PrimaryPhy,
+    ) -> None:
+        '''
+        Advertise from the REF device with the specified extended advertising
+        event properties. Use the manufacturer specific data to pad the advertising data to the
+        desired length. The scan response data must always be provided when
+        scannable.
+        '''
+        man_specific_data_length = max(0, data_len - 5)  # Flags (3) + LV (2)
+        man_specific_data = bytes([random.randint(1, 255) for _ in range(man_specific_data_length)])
+        data = DataTypes(manufacturer_specific_data=man_specific_data) if data_len > 0 else None
+        scan_response_data = None
+        # Extended advertisements with advertising data cannot also have
+        # scan response data.
+        if connectable_scannable == 'non_connectable_scannable':
+            scan_response_data = data
+            data = None
+
+        is_connectable = True if connectable_scannable == 'connectable' else False
+        target = self.dut.address if directed == 'directed' else None
+
+        # For a better test, make the secondary phy the same as the primary to
+        # avoid the scan just scanning the 1M advertisement when the primary
+        # phy is CODED.
+        secondary_phy = SECONDARY_1M
+        if primary_phy == PRIMARY_CODED:
+            secondary_phy = SECONDARY_CODED
+
+        advertise = self.ref.host.Advertise(
+            legacy=False,
+            connectable=is_connectable,
+            data=data,  # type: ignore[arg-type]
+            scan_response_data=scan_response_data,  # type: ignore[arg-type]
+            public=target,
+            own_address_type=PUBLIC,
+            primary_phy=primary_phy,
+            secondary_phy=secondary_phy,
+        )
+
+        scan = self.dut.host.Scan(
+            legacy=False,
+            passive=False,
+            timeout=self.scan_timeout,
+            phys=[primary_phy],
+        )
+        report = next((x for x in scan if x.public == self.ref.address))
+        try:
+            report = next((x for x in scan if x.public == self.ref.address))
```

Comment:
Why do we iterate two times here ?

Reply:
No reason, fixed. I think this was copied from the test_scan case, which iterates two times.